certgen/certgen.go

```diff
@@ -83,6 +83,43 @@ type CA struct {
 // in the SigningCert and SigningKey fields of the CA, and configured using the
 // CertConfig in the CA.
 func (ca *CA) IssueCertificate() (*x509.Certificate, error) {
+	return ca.issueCertificate(false)
+}
+
+// IssueCertificateChain creates a new certificate chain, containing a new leaf
+// certificate (as created by IssueCertificate) and the certificate for the key
+// that signed it (stored in the SigningCert field of the CA).
+func (ca *CA) IssueCertificateChain() ([]*x509.Certificate, error) {
+	leaf, err := ca.IssueCertificate()
+	if err != nil {
+		return nil, fmt.Errorf("error issuing leaf certificate: %s", err)
+	}
+
+	return []*x509.Certificate{leaf, ca.SigningCert}, nil
+}
+
+// IssuePrecertificate creates a new leaf precertificate, issued by the key
+// specified in the SigningCert and SigningKey fields of the CA, and configured
+// using the CertConfig in the CA.
+func (ca *CA) IssuePrecertificate() (*x509.Certificate, error) {
+	return ca.issueCertificate(true)
```

Comment:
It's nice to comment the meaning of these bare bool vals, e.g. `issueCertificate(true /*precert*/)`